aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AuthEnvironment.kt

```diff
@@ -33,14 +35,57 @@ import java.util.Date
 import java.util.UUID
 
 internal class AuthEnvironment internal constructor(
+    val context: Context,
     val configuration: AuthConfiguration,
     val cognitoAuthService: AWSCognitoAuthServiceBehavior,
     val credentialStoreClient: StoreClientBehavior,
     private val userContextDataProvider: UserContextDataProvider? = null,
     val hostedUIClient: HostedUIClient?,
     val logger: Logger
 ) : Environment {
+
+    companion object {
+        /*
+        Auth plugin needs to read from Pinpoint shared preferences, but we don't currently have an architecture
+        that allows the plugins to pass data between each other. We are duplicating this suffix constant because it
+        is internal to the Pinpoint class, which analytics does not pull in. If the Pinpoint suffix is updated, this
+        needs updated as well.
+         */
+        const val PINPOINT_SHARED_PREFS_SUFFIX = "515d6767-01b7-49e5-8273-c8d11b0f331d"
```

Comment:
would be good to add this const in core so both modules use the same value.

Reply:
I believe we may soon need an aws-core module hold things specific to aws. It doesn't belong in Core because this is Pinpoint related and ideally, all things core should be abstract.